src/systems/thermal/ThermalSensor.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "ThermalSensor.hh"
+
+#include <limits>
+#include <string>
+
+#include <ignition/common/Util.hh>
+#include <ignition/plugin/Register.hh>
+
+#include "ignition/gazebo/components/ThermalCamera.hh"
+#include "ignition/gazebo/components/Temperature.hh"
+#include "ignition/gazebo/components/TemperatureRange.hh"
+#include "ignition/gazebo/EntityComponentManager.hh"
+#include "ignition/gazebo/Util.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+/// \brief Private Thermal sensor data class.
+class ignition::gazebo::systems::ThermalSensorPrivate
+{
+};
+
+//////////////////////////////////////////////////
+ThermalSensor::ThermalSensor() : System(),
+    dataPtr(std::make_unique<ThermalSensorPrivate>())
+{
+}
+
+//////////////////////////////////////////////////
+ThermalSensor::~ThermalSensor() = default;
+
+//////////////////////////////////////////////////
+void ThermalSensor::Configure(const Entity &_entity,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    gazebo::EntityComponentManager &_ecm,
+    gazebo::EventManager & /*_eventMgr*/)
+{
+  const std::string resolutionTag = "resolution";
+  const std::string minTempTag = "min_temp";
+  const std::string maxTempTag = "max_temp";
+
+  auto sensorComp = _ecm.Component<components::ThermalCamera>(_entity);
+  if (sensorComp == nullptr)
+  {
+    ignerr << "The thermal sensor system can only be attached to a "
+           << "thermal sensor" << std::endl;
```

Comment:
is this trace right ?

Reply:
Yeah. Perhaps the wording is a little confusing, but the thermal sensor system allows users to customize the configuration/parameters of the thermal sensor/camera it's attached to. Would you recommend using different wording to help minimize confusion?

Reply:
c830f4561218d84f440b88931b3a92f94b9d8323